pulsar-client/src/main/java/org/apache/pulsar/client/impl/SameAuthParamsLookupAutoClusterFailover.java

```diff
@@ -123,7 +130,7 @@ private int firstHealthyPulsarService() {
     }
 
     private int findFailoverTo() {
-        for (int i = currentPulsarServiceIndex + 1; i <= pulsarServiceUrlArray.length; i++) {
+        for (int i = currentPulsarServiceIndex + 1; i < pulsarServiceUrlArray.length; i++) {
```

Comment:
Just wondering if this logic is correct in the first place. Shouldn't this wrap around the array?

the for loop in `checkPulsarServices` looks strange too.

Reply:
The logic of fallover is follows
- if the current cluster is the highest priority cluster and it is healthy, do nothing.
- else if there is a healthy and high priority cluster, recover to the higher priority one.
- else if both current cluster and higher priority clusters are not healthy, find a healthy cluster to fallover to.

So the two method `checkPulsarServices` and `findFailoverTo` are for the different cases to same CPU resouces.
- `checkPulsarServices` is for checking whether their is a healthy and high priority cluster.
- `findFailoverTo` is for checking whether their has a cluster is healthy.

Reply:
> what happens in the case where currentPulsarServiceIndex is already the last one? How can it find a failover cluster in that case?

It will transfer to the first healthy cluster. Once encounters the `ArrayIndexOutOfBoundsException` error, it means no cluster is healthy.

The issue the correct PR solved affects nothing